addons/editor-number-arrow-keys/userscript.js

```diff
@@ -0,0 +1,66 @@
+export default async function ({ addon }) {
+  const settings = {
+    none: 0,
+    tenth: 0.1,
+    one: 1,
+    ten: 10,
+  };
+
+  const amountOfDecimals = (num) => {
+    if (num % 1 === 0) return 0;
+    return num.toString().split(".")[1].length;
+  };
+
+  const shiftDecimalPointToRight = (num, times) => {
+    const numberIsNegative = num[0] === "-";
+    let numStr = num.toString();
+    if (numberIsNegative) numStr = numStr.substring(1);
+    for (let i = 0; i < times; i++) {
+      if (numStr.indexOf(".") === -1) numStr += 0;
+      else if (numStr.indexOf(".") === numStr.length - 2) numStr = numStr.replace(".", "");
+      else {
+        const index = numStr.indexOf(".");
+        const numArrFiltered = Array.from(numStr.replace(".", ""));
+        numArrFiltered.splice(index + 1, 0, ".");
+        numStr = numArrFiltered.join("");
+      }
+    }
+    return Number(numStr) * (numberIsNegative ? -1 : 1);
+  };
+
+  document.body.addEventListener("keydown", (e) => {
+    if (!e.target.classList.contains("blocklyHtmlInput")) return;
+    if (!["ArrowUp", "ArrowDown"].includes(e.code)) return;
+    if (
+      e.target.value &&
+      Number(e.target.value)
+        .toString()
+        .replace(/^0*|0*$/, "") !== e.target.value.replace(/^0*|0*$/, "")
+    )
+      return;
```

Comment:
This will not return if `e.target.value` is null, so it will fail on line 42 still

Reply:
1. When would `e.target.value` be null?
2. Doesn't the simple `e.target.value` check take care of that?

Reply:
Oh, wait, yeah, true, I didn't think this through

```suggestion
    if (
      !e.target.value ||
      Number(e.target.value)
        .toString()
        .replace(/^0*|0*$/, "") !== e.target.value.replace(/^0*|0*$/, "")
    )
      return;
```

Reply:
> 1. When would `e.target.value` be null?

Well, I meant falsey. It might also be an empty string

Reply:
An empty string was what I was originally trying to prevent, but I messed up the logic.